### ISSUE TITLE
OpenSSL testing to confirm OpenSSL natives are being loaded

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#4092]
As per https://github.com/eclipse/openj9/issues/3636#issuecomment-448048339, OpenSSL support may be silently failing. There should be a test, likely using the `-Djdk.nativeCryptoTrace=true` option, to confirm the natives are being loaded as expected.

**Comment by pshipton** [comment]
@enasser @smlambert

**Comment by enasser** [comment]
@pshipton  @smlambert  A simple test could be as follows. It will throw USLE if there is a failure. In the automated testcase, we can catch the exception and assert failure.

```
public class LoadTest {
   public static void main(String[] args) throws Exception {
       System.loadLibrary("jncrypto");
   }
}
```

**Comment by enasser** [comment]
Another interesting observation is that in my x64 linux system, it did not create the dependency to libssl library even though we used -lssl. Looks like the linker ignore the dependency as no symbols from libssl is used by libjncrypto. Hence, the builds from the local system did not fail in the above test.

With AdoptOpenJDK build:
```
root@ojdk-xa641:~/testpgm# /root/build/openj9-jdk-11.0.1+13/bin/java -showversion LoadTest
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.1+13-201812171911)
Eclipse OpenJ9 VM AdoptOpenJDK (build master-3f4c738c6, JRE 11 Linux amd64-64-Bit Compressed References 20181217_62 (JIT enabled, AOT enabled)
OpenJ9   - 3f4c738c6
OMR      - 2b09f123
JCL      - f6d23ec079 based on jdk-11.0.1+13)
Exception in thread "main" java.lang.UnsatisfiedLinkError: jncrypto (libssl.so.1.1: cannot open shared object file: No such file or directory)
	at java.base/java.lang.ClassLoader.loadLibraryWithPath(ClassLoader.java:1678)
	at java.base/java.lang.ClassLoader.loadLibraryWithClassLoader(ClassLoader.java:1648)
	at java.base/java.lang.System.loadLibrary(System.java:559)
	at LoadTest.main(LoadTest.java:3)
```
With local build:
```
root@ojdk-xa641:~/testpgm# /root/openj9/orig/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/bin/java -showversion LoadTest
openjdk version "11.0.1-internal" 2018-10-16
OpenJDK Runtime Environment (build 11.0.1-internal+0-adhoc.root.openj9-openjdk-jdk11)
Eclipse OpenJ9 VM (build master-8ca674a, JRE 11 Linux amd64-64-Bit Compressed References 20181208_000000 (JIT enabled, AOT enabled)
OpenJ9   - 8ca674a
OMR      - 3a90b31
JCL      - 69393ac based on jdk-11.0.1+13)
```

**Comment by pshipton** [comment]
The OpenJ9 nightly build is the same as the local build, seems the problem only occurrs with Adopt builds.

**Comment by pshipton** [comment]
@smlambert is somebody able to look at this, this week?

**Comment by pshipton** [comment]
@smlambert what is the status of this? We have some pending openssl related changes, and no testing to confirm openssl is working before they are merged.

https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/227
https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/90
https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/236

**Comment by smlambert** [comment]
Initial suggested fix in my branch.  I need to test it and create PR.  

To clarify, this fix does not test openssl is working, it tests that the jncrypto native lib is loaded.

**Comment by pshipton** [comment]
> it tests that the jncrypto native lib is loaded

Right, that is what I meant by "working".

**Comment by pshipton** [comment]
#4300 was merged.

**Comment by pshipton** [comment]
The current test is not sufficient. With the latest changes (below), the load of jncyrpto may succeed but this doesn't indicate if the openssl support can work. We need to check via `-Djdk.nativeCryptoTrace=true`

https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/227
https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/90

**Comment by pshipton** [comment]
Looking again at the current test, it contains `expectedExceptions = UnsatisfiedLinkError.class` so I'm not sure that it tests anything.

**Comment by smlambert** [comment]
When I initially ran this test, since the framework loads the library, this test throws ULE with message already loaded.  But as we had discussed earlier, a cmdlinetester tests that looks for correct trace msgs is better.

It sounds like the implementation has changed somewhat also, loading one thing, then later loading other stuff, so more testing is good.  I intentionally created a separate class and separate test target with the idea we could expand on testing.

**Comment by pshipton** [comment]
@llxia The jdk.nativeCryptoTrace output is as follows. We expect "using Native crypto library" for success and "Native crypto library load failed" for failure.

https://github.com/ibmruntimes/openj9-openjdk-jdk8/blob/2bf413128ab7cbe9d565cc16ffb1a1052b7bbcb2/jdk/src/share/classes/com/sun/crypto/provider/CipherCore.java#L1329

https://github.com/ibmruntimes/openj9-openjdk-jdk8/blob/2bf413128ab7cbe9d565cc16ffb1a1052b7bbcb2/jdk/src/share/classes/sun/security/provider/SunEntries.java#L445

**Comment by pshipton** [comment]
`KeyFactory.getInstance("RSA");` is sufficient to cause the pass / fail output in CipherCore. Since both cases use NativeCrypto and load a single native library, it should be sufficient to test this.

**Comment by smlambert** [comment]
Fixed by #4376 added test excluded label to find this issue later and enable for win/osx/aix
